ops/testing.py

```diff
@@ -424,7 +424,7 @@ def remove_relation(self, relation_id: int) -> None:
         except KeyError as e:
             raise model.RelationNotFoundError from e
 
-        for unit_name in self._backend._relation_list_map[relation_id]:
+        for unit_name in self._backend._relation_list_map[relation_id].copy():
```

Comment:
I wonder if '.keys()' would be better here, which is still a copy but doesn't have to copy any of the values.